### ISSUE TITLE
Validation split for MNIST in Fuel

### ISSUE THREAD

**Issue by rizar** [mila-udem/fuel#306]
We should consider including "validation" split to our MNIST dataset. It has certain official status, as far as I know.

**Comment by dwf** [comment]
I don't think it does...

**Comment by dmitriy-serdyuk** [comment]
Oh my goodness.. 
- The official [page](http://yann.lecun.com/exdb/mnist/) doesn't define any validation split
- deeplearning.net offers to use 10k examples for [validation](http://deeplearning.net/tutorial/gettingstarted.html) and seems, that all theano based libraries use this split
- [Tensor flow](https://www.tensorflow.org/versions/0.6.0/tutorials/mnist/download/index.html) uses only 5k examples

**Comment by rizar** [comment]
But I am pretty sure that in many well-reputed paper the last 10000
examples are used (too lazy to go ahead and check it right now).

**Comment by vdumoulin** [comment]
It's a matter of trading between convenience and safety. I'm afraid that this might lead to accidentally contaminating the training set with validation set examples (e.g. using the full official training set (60,000 examples) and the unofficial validation set (last 10,000 examples of the training set)).

If you think it would be really convenient to have the unofficial train/valid split as opposed to having to specify a `subset` argument to the constructor, I'd suggest that we choose the split names to be unambiguous (`unofficial_train` and `unofficial_valid`, for instance) and use the `MNIST` constructor to alias these to a superclass constructor call with the proper `subset` kwarg.

**Comment by dwf** [comment]
It is empirically true that using the last 10k examples is good as a
validation set, much better than using a random 10k which will cause early
stopping to massively overfit in my experience. But it is somewhat
mysterious why this is

**Comment by rizar** [comment]
At the risk of revealing my ignorance, I will dare to ask: can't we have
overlapping splits in Fuel? Can we have "unofficial_train",
"unofficial_valid" as additional splits?

**Comment by vdumoulin** [comment]
No, you're right. It would involve changing the converter as well as the dataset class, but it's doable. It all depends on where you want to bake in these additional splits. Actually, doing so at the converter level might be a more elegant solution.

**Comment by rizar** [comment]
I think this will definitely contribute to reproducibility of the research
at the lab.. :) Let's make it a CCW ticket?
